raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1115,7 +1126,9 @@ private EndQuorumEpochRequestData buildEndQuorumEpochRequest() {
         return new EndQuorumEpochRequestData()
                 .setReplicaId(quorum.localId)
                 .setLeaderId(quorum.leaderIdOrNil())
-                .setLeaderEpoch(quorum.epoch());
+                .setLeaderEpoch(quorum.epoch())
+                .setPreferredSuccessors(
+                    quorum.leaderStateOrThrow().nonLeaderVotersByDescendingFetchOffset());
```

Comment:
Does it make sense to limit this list to followers above the high watermark?

Reply:
I feel having the entire list helps validating the EndQuorumRequest itself is sent from a leader consistent with our state.

Reply:
I'd agree with @abbccdda here to just list all voters, since if a voter found itself not in the list it would be a bit tricky how to set its backoff before starting the vote.